examples/ambassador is not working

## Bug

### Current Behavior

`garden deploy` does not work with `examples/ambassador`

### Expected behavior

`garden deploy` should work as documented in the `README`

### Reproducible example

Using `minikube` on OSX

```bash
# On a clean k8s cluser
minikube start

# go to examples/ambassador
garden deploy
```
Outputs the following:

```
Failed deploying service 'ambassador' (from module 'ambassador'). Here is the output:
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Command "/Users/biscii/.garden/tools/helm3/dd4668d875bedaf5/darwin-amd64/helm --kube-context minikube --namespace ambassador install ambassador 
/Users/biscii/Development/open-source/garden-demo/examples/ambassador/.garden/build/ambassador/ambassador --atomic --namespace ambassador --timeout 300s 
--values /Users/biscii/Development/open-source/garden-demo/examples/ambassador/.garden/build/ambassador/ambassador/garden-values.yml" failed with code 1:
coalesce.go:199: warning: destination for annotations is a table. Ignoring non-table value <nil>
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
Error: release ambassador failed, and has been uninstalled due to atomic being set: timed out waiting for the condition
coalesce.go:199: warning: destination for annotations is a table. Ignoring non-table value <nil>
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
manifest_sorter.go:175: info: skipping unknown hook: "crd-install"
Error: release ambassador failed, and has been uninstalled due to atomic being set: timed out waiting for the condition
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```


### Suggested solution(s)

1. The chart is deprecated and CRD/s need to be handled differently. Unsure how best to implement within the garden build context.

- https://github.com/helm/charts/tree/master/stable/ambassador

2. Update references in docs from `custom-ingress-controller` to `ambassador`

### Your environment

`garden version` - 0.11.9

`kubectl version`

```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:16:51Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

`docker version`

```
Client: Docker Engine - Community
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        afacb8b
 Built:             Wed Mar 11 01:21:11 2020
 OS/Arch:           darwin/amd64
 Experimental:      false


Server: Docker Engine - Community
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       afacb8b
  Built:            Wed Mar 11 01:29:16 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

--- a/examples/ambassador/ambassador/garden.yml
+++ b/examples/ambassador/ambassador/garden.yml
@@ -2,16 +2,16 @@ kind: Module
 description: Ambassador API Gateway
 type: helm
 name: ambassador
-chart: stable/ambassador
+repo: https://getambassador.io
+chart: ambassador
+version: "6.2.2"
 values:
   service:
-    annotations:
-      getambassador.io/config: |
-        ---
-        apiVersion: ambassador/v1
-        kind: Module
-        name: ambassador
-        config:
-          service_port: 8080 # Set port since the default ingress already occupies the default port
-    http:
-      port: 8080 # Set port since the default ingress already occupies the default port
+    # Setting non-default HTTP ports to avoid conflict with other ingress controllers in the cluster
+    ports:
+      - name: http
+        port: 8080
+        targetPort: 8080
+      - name: https
+        port: 8443
+        targetPort: 8443

